### PR TITLE
feat(api): Enable partialProjectName search for custom-projects

### PR DESCRIPTION
**Pull request by alepefe**
This pull request introduces enhancements to the `CustomProjectsService` by adding new filtering capabilities and updating the associated schema and tests. The most important changes include the introduction of a new schema for query parameters, modifications to the service to support partial project name filtering, and the addition of a test case to verify the new functionality.

### Schema and Type Updates:
* Added `partialProjectName` to the `customProjecsQuerySchema` in `shared/contracts/custom-projects.contract.ts` to enable filtering by partial project name.
* Defined a new type `CustomProjectFetchSpecificacion` based on the updated schema in `api/src/modules/custom-projects/custom-projects.service.ts`.

### Service Enhancements:
* Updated the `extendFindAllQuery` method in `CustomProjectsService` to use the new `CustomProjectFetchSpecificacion` type and added logic to filter projects by `partialProjectName`. [[1]](diffhunk://#diff-a450b42f83aa743e906c235e1dbcc649854328c6951985e9bcaf6a40b6029eacL124-R129) [[2]](diffhunk://#diff-a450b42f83aa743e906c235e1dbcc649854328c6951985e9bcaf6a40b6029eacR138-R143)

### Testing:
* Added a new test case in `api/test/integration/custom-projects/custom-projects-read.spec.ts` to ensure that authenticated users can retrieve custom projects filtered by `partialProjectName`.

- [Figma profile page](https://www.figma.com/design/ZzvgiambS60zL5oVbMgcyF/%5BWIP%5D---Blue-Carbon-tool-Visuals?node-id=5810-44466&m=dev)
- [JIRA ticket](https://vizzuality.atlassian.net/browse/TBCCT-147)

### CODE REVIEW

File: client/src/lib/query-keys.ts [hunk 1]
```diff
@@ -45,6 +45,12 @@ export const tableKeys = createQueryKeys("tables", {
 
 export const customProjectKeys = createQueryKeys("customProjects", {
   cached: null,
+  all: <TFilters extends z.ZodTypeAny>(
+    filters?: z.infer<TFilters> & {
+      sorting?: SortingState;
+      pagination?: PaginationState;
+    },
+  ) => ["all", , filters],
```
Comment (andresgnlez) [inline client/src/lib/query-keys.ts:53]:
```suggestion
  ) => ["all" , filters],
```

Comment (andresgnlez) [inline client/src/lib/query-keys.ts:53]:
```suggestion
  ) => ["all", filters],
```

File: client/src/app/my-projects/url-store.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  activity: [],
+  keyword: "",
+};
+
+export function useMyProjectsFilters() {
+  return useQueryState(
+    "filters",
+    parseAsJson(filtersSchema.parse).withDefault(INITIAL_FILTERS_STATE),
+  );
+}
```
Comment (andresgnlez) [inline client/src/app/my-projects/url-store.ts:20]: Let's keep it but I don't see any value of keeping the state of the page in the URL in a page that only the user can access. The page itself is only accessible by the user, so making it shareable could have been optional in this case.

File: client/src/containers/my-projects/index.tsx [hunk 3]
```diff
@@ -26,44 +32,63 @@ import {
...
+        pageSize: pagination.pageSize,
+      },
+      extraHeaders: {
+        authorization: `Bearer ${session?.accessToken as string}`,
+      },
+    },
+    {
+      select: (d) => d.body,
+      queryKey,
+    },
```
Comment (andresgnlez) [inline client/src/containers/my-projects/index.tsx:86]: @atrincas can you try adding [keepPreviousData](https://github.com/Vizzuality/tnc-blue-carbon-cost-tool/blob/4028cf98ddf4fc2d78d646cd2da09c54d81c8fc0/client/src/containers/overview/table/view/overview/index.tsx#L95)? This way the results don't blink whenever the user is typing.

File: client/src/containers/my-projects/columns.tsx [hunk 4]
```diff
@@ -69,33 +76,33 @@ export const columns: CustomColumn[] = [
...
-              ? "border-sky-300 bg-blue-500/20 text-sky-blue-300"
-              : "border-mint-green-200 bg-green-500/20 text-mint-green-200",
-          )}
+          className={cn({
+            "rounded-full border px-2 py-1 text-xs font-medium": true,
+            "border-sky-300 bg-blue-500/20 text-sky-blue-300":
+              getValue() === ACTIVITY.CONSERVATION,
+            "border-mint-green-200 bg-green-500/20 text-mint-green-200":
+              getValue() === ACTIVITY.RESTORATION,
+          })}
```
Comment (andresgnlez) [inline client/src/containers/my-projects/columns.tsx:105]: @atrincas we already have a `Badge` component for this, can you please check?

File: client/src/app/my-projects/url-store.ts (hunk 2 above)
Comment (atrincas) [reply client/src/app/my-projects/url-store.ts:20]: A user can bookmark his filters in the browser, other than that I agree with you

**Review by andresgnlez — APPROVED** [review]

**Pull request merged**